ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/part/PartStackViewImpl.java

```diff
@@ -40,43 +54,162 @@
+                delegate.onToggleMaximize();
+            }
+        });
+
+        if (maximizeButton.getElement() instanceof elemental.dom.Element) {
+            Tooltip.create((elemental.dom.Element) maximizeButton.getElement(),
+                    PositionController.VerticalAlign.BOTTOM, PositionController.HorizontalAlign.MIDDLE, localizationConstant.maximizePartStackTitle());
+        }
+    }
+
+    /**
+     * Adds button to hide part stack.
+     */
+    private void addHideButton() {
+        SVGImage hide = new SVGImage(resources.collapseExpandIcon());
+        hide.getElement().setAttribute("name", "workBenchIconMinimize");
+        ToolButton hideToolButton = new ToolButton(hide);
+        hideButton.add(hideToolButton);
+
+        hideToolButton.addClickHandler(new ClickHandler() {
+            @Override
+            public void onClick(ClickEvent event) {
+                delegate.onHide();
+            }
+        });
+
+        if (hideButton.getElement() instanceof elemental.dom.Element) {
+            Tooltip.create((elemental.dom.Element) hideButton.getElement(),
+                    PositionController.VerticalAlign.BOTTOM, PositionController.HorizontalAlign.MIDDLE, localizationConstant.minimizePartStackTitle());
+        }
+    }
+
+    /**
+     * Adds part stack options button.
+     */
+    private void addMenuButton() {
+        final ToolButton menuToolButton = new ToolButton(FontAwesome.COG);
+        menuToolButton.getElement().setAttribute("name", "workBenchIconPartStackOptions");
+        menuButton.add(menuToolButton);
+        menuToolButton.addClickHandler(new ClickHandler() {
+            @Override
+            public void onClick(ClickEvent event) {
+                Scheduler.get().scheduleDeferred(() -> {
+                    int left = getAbsoluteLeft(menuToolButton.getElement());
+                    int top = getAbsoluteTop(menuToolButton.getElement());
+                    delegate.onPartStackMenu(left + 10, top + 21 - 8);
+                });
+            }
+        });
+
+        if (menuButton.getElement() instanceof elemental.dom.Element) {
+            Tooltip.create((elemental.dom.Element) menuButton.getElement(),
+                    PositionController.VerticalAlign.BOTTOM, PositionController.HorizontalAlign.MIDDLE, localizationConstant.partStackOptionsTitle());
+        }
+    }
+
+    /**
+     * Returns absolute left position of the element.
+     *
+     * @param element
+     *          element
+     * @return
+     *          element left position
+     */
+    private native int getAbsoluteLeft(JavaScriptObject element) /*-{
+        return element.getBoundingClientRect().left;
+    }-*/;
+
+    /**
+     * Returns absolute top position of the element.
+     *
+     * @param element
+     *          element
+     * @return
+     *          element top position
+     */
+    private native int getAbsoluteTop(JavaScriptObject element) /*-{
+        return element.getBoundingClientRect().top;
```

Comment:
I'd rather move this method to DomUtils, it's an utility method which may use by anyone.